config/initializers/gds-api-adapters.rb

```diff
@@ -0,0 +1 @@
+GdsApi.config.always_raise_for_not_found = true
```

Comment:
Maybe we do. TBH, this PR is for general upgrade maintenance, separate from the story about the adapters.  There's work to be done to get it up to the most recent api adapters version too (see: https://github.com/alphagov/licence-finder/pull/82/commits/2cabcac0706c02c1392259a169b9fbb0cb418eb2).